packages/puppeteer-core/src/bidi/Target.ts

```
@@ -4,149 +4,129 @@
-    browsingContext: BrowsingContext
-  ) {
-    super(browserContext);
-
-    this._browsingContext = browsingContext;
+  constructor(frame: BidiFrame) {
+    super();
+    this.#frame = frame;
   }
 
+  override async page(): Promise<BidiPage> {
+    if (this.#page === undefined) {
+      this.#page = BidiPage.from(
+        this.browserContext(),
+        this.#frame.browsingContext
+      );
+    }
+    return this.#page;
+  }
+  override async asPage(): Promise<BidiPage> {
+    return BidiPage.from(this.browserContext(), this.#frame.browsingContext);
+  }
   override url(): string {
-    return this._browsingContext.url;
+    return this.#frame.url();
   }
-
-  override async createCDPSession(): Promise<CDPSession> {
-    const {sessionId} = await this._browsingContext.cdpSession.send(
-      'Target.attachToTarget',
-      {
-        targetId: this._browsingContext.id,
-        flatten: true,
-      }
-    );
-    return new BidiCdpSession(this._browsingContext, sessionId);
+  override createCDPSession(): Promise<CDPSession> {
+    return this.#frame.createCDPSession();
   }
-
   override type(): TargetType {
     return TargetType.PAGE;
   }
-}
-
-/**
- * @internal
- */
-export class BiDiPageTarget extends BiDiBrowsingContextTarget {
-  #page: BidiPage;
-
-  constructor(
-    browserContext: BidiBrowserContext,
-    browsingContext: BrowsingContext
-  ) {
-    super(browserContext, browsingContext);
-
-    this.#page = new BidiPage(browsingContext, browserContext);
+  override browser(): BidiBrowser {
+    return this.browserContext().browser();
   }
-
-  override async page(): Promise<BidiPage> {
-    return this.#page;
+  override browserContext(): BidiBrowserContext {
+    return this.#frame.page().browserContext();
   }
-
-  override _setBrowserContext(browserContext: BidiBrowserContext): void {
-    super._setBrowserContext(browserContext);
-    this.#page._setBrowserContext(browserContext);
+  override opener(): Target | undefined {
+    return;
```

let's throw UnsupportedError